### PR TITLE
Support custom version in installation via .sh script

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Added support for CLI version in insall.sh

## Why?

Used by https://github.com/temporalio/docker-builds/pull/112

## Checklist

1. Closes 

2. How was this tested:

Manually so far. There is an install.test.sh file for testing, i'd prefer to have testing with a proper language (not done).

```
sh -c "$(cat ./install.sh)" -- --dir ./temp
(cd ./temp/bin && ./temporal -h)
...
VERSION:
   0.9.0 (server 1.20.2) (ui 2.15.0)
```

```
sh -c "$(cat ./install.sh)" -- --dir ./temp --version v0.3.0
(cd ./temp/bin && ./temporal -h)
...
VERSION:
   0.3.0 (server 1.19.1) (ui 2.9.1)
```

```
sh -c "$(cat ./install.sh)" -- --dir ./temp --version 0.3.0
(cd ./temp/bin && ./temporal -h)
...
VERSION:
   0.3.0 (server 1.19.1) (ui 2.9.1)
```

3. Any docs updates needed?

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -347,6 +352,32 @@ get_install_dir() {
...
+    local _version
+    _version="latest"
+
+    while [[ $# -gt 0 ]]; do
+        case "$1" in
+        --version)
+            _version="$2"
+            shift 2
+            ;;
+        *)
```
Comment (dnr) [inline install.sh:368]: This and the loop in `get_install_dir` should be written the same way. I like this way better but either way, be consistent

File: install.sh [hunk 2]
```diff
@@ -347,6 +352,32 @@ get_install_dir() {
...
+            shift
+            ;;
+        esac
+    done
+
+    if [[ $_version == v* ]] || [[ $_version == "latest" ]]; then
+        printf %s "$_version"
+    else
+        printf 'v%s' "$_version"
+    fi
```
Comment (dnr) [inline install.sh:378]: this reads better to me if you do it like
```suggestion
    if [[ $_version == [0-9]* ]]; then
        _version="v$_version"
    fi
    printf %s "$_version"
```

**Review by dnr — APPROVED** [review]
couple nitpicks but with those it's good

File: install.sh (hunk 1 above)
Comment (feedmeapples) [reply install.sh:368]: updated

File: install.sh (hunk 2 above)
Comment (feedmeapples) [reply install.sh]: used your logic + few changes to address shellcheck

File: install.sh [hunk 3]
```diff
@@ -371,9 +371,10 @@ get_version() {
         esac
     done
 
-    if [[ $_version == [0-9]* ]]; then
+    if echo "$_version" | grep -qE '^[0-9]+'; then
```
Comment (dnr) [inline install.sh:374]: If `if` doesn't work, pretty sure `case` will, but this is fine too.
```suggestion
    if echo "$_version" | grep -q '^[0-9]'; then
```

**Review by dnr — APPROVED** [review]

**Pull request merged**